src/action.py

```diff
@@ -81,6 +104,9 @@ def _create(
             url = FAIL_NAVLINK_LINK
             result = types_.ActionResult.FAIL
             reason = str(exc)
+    # This should not occur since all the create action types have been implemented
+    else:  # pragma: no cover
+        raise NotImplementedError(f"internal error: {action} has not been implemented")
```

Comment:
I am guessing this is targeting python 3.10 so this does not apply, but they added an [`assert_never` in Python 3.11](https://docs.python.org/3/library/typing.html#typing.assert_never).

Reply:
Yeh this would be interesting - the code is actually compliant with 3.11 (and in production 3.11 is used) - just in `operator-workflows` we're using `22.04` which comes with 3.10 and hence the testing and linting is run on 3.10